why.trim is not a function on prettyError

https://github.com/inngest/inngest-js/blob/52398dc412d0ad2fd8fabc40638537014db3f8b0/packages/inngest/src/helpers/functions.ts#L215C26-L215C31

This likely causes an issue where `why` is treated as a string, while it isn't. 

```
[1] ✘ [ERROR] TypeError: why.trim is not a function
[1] 
[1]       at prettyError (/node_modules/src/helpers/errors.ts:394:41)
[1]       at fetchAllFnData (/node_modules/src/helpers/functions.ts:212:11)
[1]       at null.<anonymous> (async /hypervisor/.wrangler/tmp/dev-tSuJSe/dev.js:13902:29)
[1]       at async InngestCommHandler.handleAction (/node_modules/src/components/InngestCommHandler.ts:733:28)
[1]       at async ServerTiming.wrap (/node_modules/src/helpers/ServerTiming.ts:83:15)
[1]       at async ServerTiming.wrap (/node_modules/src/helpers/ServerTiming.ts:83:15)
[1]       at async Object.fetch (/hypervisor/src/dev.ts:70:12)
[1]       at async jsonError (/node_modules/wrangler/templates/middleware/middleware-miniflare3-json-error.ts:22:10)
[1] 
```

Reference to discord discussion: https://discord.com/channels/842170679536517141/1206012623817146458

--- a/.changeset/mighty-grapes-sell.md
+++ b/.changeset/mighty-grapes-sell.md
@@ -1,0 +1,5 @@
+---
+"inngest": patch
+---
+
+Fix API failures being masked at runtime when fetching large step data or event batches fails

--- a/packages/inngest/src/api/api.ts
+++ b/packages/inngest/src/api/api.ts
@@ -1,6 +1,7 @@
 import { type fetch } from "cross-fetch";
 import { type ExecutionVersion } from "../components/execution/InngestExecution";
 import { getFetch } from "../helpers/env";
+import { getErrorMessage } from "../helpers/errors";
 import { hashSigningKey } from "../helpers/strings";
 import { err, ok, type Result } from "../types";
 import {
@@ -65,7 +66,10 @@ export class InngestApi {
         }
       })
       .catch((error) => {
-        return err({ error: error as string, status: 500 });
+        return err({
+          error: getErrorMessage(error, "Unknown error retrieving step data"),
+          status: 500,
+        });
       });
   }
 
@@ -87,7 +91,10 @@ export class InngestApi {
         }
       })
       .catch((error) => {
-        return err({ error: error as string, status: 500 });
+        return err({
+          error: getErrorMessage(error, "Unknown error retrieving event batch"),
+          status: 500,
+        });
       });
   }
 }

--- a/packages/inngest/src/helpers/errors.ts
+++ b/packages/inngest/src/helpers/errors.ts
@@ -342,6 +342,19 @@ const isError = (err: unknown): err is Error => {
 };
 
 /**
+ * Given an `unknown` object, retrieve the `message` property from it, or fall
+ * back to the `fallback` string if it doesn't exist or is empty.
+ */
+export const getErrorMessage = (err: unknown, fallback: string): string => {
+  const { message } = z
+    .object({ message: z.string().min(1) })
+    .catch({ message: fallback })
+    .parse(err);
+
+  return message;
+};
+
+/**
  * Given a {@link PrettyError}, return a nicely-formatted string ready to log
  * or throw.
  *

